synapse/storage/data_stores/main/devices.py

```diff
@@ -129,6 +131,33 @@ def get_devices_by_remote(self, destination, from_stream_id, limit):
         if not updates:
             return now_stream_id, []
 
+        # get the cross-signing keys of the users the list
+        users = set(r[0] for r in updates)
+        master_key_by_user = {}
+        self_signing_key_by_user = {}
+        for user in users:
+            cross_signing_key = yield self.get_e2e_cross_signing_key(user, "master")
+            if cross_signing_key:
+                key_id, verify_key = get_verify_key_from_cross_signing_key(
+                    cross_signing_key
+                )
+                master_key_by_user[user] = {
+                    "key_info": cross_signing_key,
+                    "pubkey": verify_key.version,
```

Comment:
it's a bit confusing that the "version" is also called "pubkey" and then used as a device id.

I'd suggest renaming this field to `device_id` and adding a comment to explain why it's correct to compare `version` with device_ids.